torch/csrc/jit/serialization/import.h

```diff
@@ -17,27 +17,46 @@ class ReadAdapterInterface;
 namespace torch {
 namespace jit {
 
-// used in torch.package deserialization
+// Used in torch.package and TorchScript serialization and
+// deserialization to coordinate sharing of storages between
+// models. Creates mapping between a 'name', a unique numerical
+// ID for that name, and a c10::Storage reference. The add/has/get
+// methods operate with the name as keys and not with the unique
+// ids as keys. The unique ID is included to aid in creating
+// deterministic naming for storages during serialization.
 class TORCH_API StorageContext {
  public:
-  explicit StorageContext() = default;
-
-  void addStorage(const std::string& name, c10::Storage storage) {
-    storage_map_.insert({name, storage});
+  explicit StorageContext() : unique_id(0) {}
+
+  uint64_t addStorage(const std::string& name, c10::Storage storage) {
```

Comment:
I think the API for this can be made simpler by removing the concept of names. 
- addStorage(storage) -> id
- hasStorage(id) -> bool
- getId(storage) -> id
- getStorage(id) -> storage

We don't really need names, right? Since we plan to serialize the id anyway.

Reply:
The names help solve a weird situation the deserialization process. Basically during unpicking we have the storage’s ‘name’ returned from the storage’s persid tuple and that’s it. In the `StorageContext` we need a way for both the TS unpickler and torch.package’s load_tensor function to be able to map this outside-the-storage-context key to the same storage, and that’s not doable with the unique_id if the unique_id is being generated from the `StorageContext` itself. The weirdness of this StorageContext is really coming from us trying to use it for both serialization and deserialization which have different needs (serialization needing the unique_id and the deseralziation needing to map arbitrary names to storages)